### DISCUSSION TITLE
DNSDist wont connect to authoritive server on different port then 53

### DISCUSSION THREAD

**Discussion by StonedPower** [PowerDNS/pdns#14545] [Q&A] [answered]
I have a DNSDist and a PowerDNS Autoritive docker container running on the same server. 
When setting up the Autoritive server I gave it a port of 5300 to use, but in DNSDist this shows up as "Down", meanwhile if I change it to port 53 it shows as "up".

Is there something I'm missing why this isn't working?

**Comment by Habbie** [comment]
Different ports should work, if configured correctly on both sides. Does `setVerbose(true)` provide any useful information?

**Comment by StonedPower** [comment]
Not much in the logging I can see:
```
dnsdist 1.9.6 comes with ABSOLUTELY NO WARRANTY. This is free software, and you are welcome to redistribute it according to the terms of the GPL version 2
Added downstream server 172.18.0.2:53
Adding server to default pool
Added downstream server 172.18.0.6:1053
Adding server to default pool
Listening on 0.0.0.0:53
ACL allowing queries from: 0.0.0.0/0
Console ACL allowing connections from: 127.0.0.0/8, ::1/128
Adding TCP Client thread
Adding TCP Client thread
Adding TCP Client thread
Adding TCP Client thread
Adding TCP Client thread
Adding TCP Client thread
Adding TCP Client thread
Adding TCP Client thread
Adding TCP Client thread
Adding TCP Client thread
Adding DoH Client thread
Webserver launched on 0.0.0.0:8080
Marking downstream 172.18.0.2:53 as 'up'
Marking downstream 172.18.0.6:1053 as 'down'
Polled security status of version 1.9.6 at startup, no known issues reported: OK
```

**Reply by Habbie** [reply]
if you start a shell in the dnsdist container, can dig reach port 1053 on that IP?

**Reply by StonedPower** [reply, marked as answer]
I don't know how, but I found the solution. 

Within portainer the Authoritive server uses port 1053 on the "outside" and 53 inside the container:
![image](https://github.com/user-attachments/assets/3a7c96f6-cbcf-43a2-a96a-d5b020d8aad7)

But in the DNSDist config file, I don't have to supply pot 1053 but rather nothing and let it use port 53: 
![image](https://github.com/user-attachments/assets/20aa9195-06f5-4f4d-9247-a15e4e775da8)

And looking at the webserver is shows as UP:
![image](https://github.com/user-attachments/assets/3138af39-8730-47ae-a8a8-f59146c53c50)

Doing a nslookup on my laptop gives me the right results: 
![image](https://github.com/user-attachments/assets/88a9284b-3f9d-4513-802f-709985c1cebd)

I don't know why this is working, but its working, Thanks Habbie for the support!

**Comment by StonedPower** [comment]
Here is my dnsdist.conf:  _(This is just a test setup and not publicly accessible)_
```
setACL("0.0.0.0/0")
setLocal('0.0.0.0')

setVerbose(true)


newServer{address="172.18.0.2"}
newServer{address="172.18.0.6:1053"}
setServerPolicy(firstAvailable)

webserver("0.0.0.0:8080")
setWebserverConfig({password="<PW>", apiKey="<PW>", acl="0.0.0.0/0"})

```
